modules/presets/field.js

```diff
@@ -10,6 +10,11 @@ export function presetField(fieldID, field, allFields) {
   allFields = allFields || {};
   let _this = Object.assign({}, field);   // shallow copy
 
+  // This handles fieldIDs that contain ':', like 'source:1'
+  let localizerFieldID = fieldID.includes(':') ? fieldID.split(':')[0] + '_multiple': fieldID;
+  // This is what comes after the ':' in the fieldID. 
+  let localizerOption = fieldID.includes(':') ? fieldID.split(':')[1]: 0;
```

Comment:
The automatic number formatting only works if the option value is a number (i.e., passing it into `parseInt()`). This would be pretty specific to `source:*` or any other key we define in the future that uses numeric indices. Also, I’m not sure if we can be sure that any field ID containing a colon puts a number after the colon.

Maybe the field definition in index.js could include a property that says the key can have a numeric “index” suffix. Then this code could automatically split it out and convert it to a number when it sees that property. Or instead of introducing a new property, you could at least check whether the `type` is `source`.

Reply:
This makes sense! I decided to create two properties in the multiple source fields: `field.baseKey`, which is 'source' and `field.index`, which is and integer defined in index.js. Those values are used to calculate `localizerFieldID`.